ssds/cli/staging.py

```diff
@@ -26,9 +25,8 @@ def upload(args: argparse.Namespace):
     """
     ssds = Staging[args.deployment].ssds
     root = os.path.abspath(os.path.normpath(args.path))
-    with ThreadPoolExecutor(max_workers=4) as e:
-        for ssds_key in ssds.upload(root, args.submission_id, args.name, e):
-            print(ssds.compose_blobstore_url(ssds_key))
+    for ssds_key in ssds.upload(root, args.submission_id, args.name, threads=3):
```

Comment:
Was `workers` -> `threads` downgraded from 4 to 3?  Should this be a function of the system hardware?

Reply:
This is IO bound, not CPU bound. I've noticed that 3 is working better than 4, but I'm not sure how to adjust this number automatically.